File: demo/Blackholio/client-unity/Assets/Scripts/DeathScreen.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class DeathScreen : MonoBehaviour
{
	public Button RespawnButton;

	public void SetVisible(bool visible)
	{
		gameObject.SetActive(visible);
	}

	public void Respawn()
	{
		GameManager.Conn.Reducers.Respawn();
		SetVisible(false);
	}
}

File: demo/Blackholio/client-ts/src/ui/DeathScreen.ts
export class DeathScreen {
  private readonly overlay = document.querySelector(
    '#death-overlay'
  ) as HTMLElement;
  private readonly respawnButton = document.querySelector(
    '#respawn-button'
  ) as HTMLButtonElement;

  constructor(onRespawn: () => void) {
    this.respawnButton.addEventListener('click', onRespawn);
  }

  setVisible(visible: boolean): void {
    this.overlay.classList.toggle('hidden', !visible);
  }
}
